fvm/transactionInvoker.go

```diff
@@ -49,31 +49,41 @@ func (i TransactionInvoker) Process(
 
 	var modifiedSets programsCache.ModifiedSetsInvalidator
 	defer func() {
-		// based on the contract and frozen account updates we decide how to
-		// clean up the programs for failed transactions we also do the same as
-		// transaction without any deployed contracts
-		programs.Cleanup(modifiedSets)
-
 		if sth.NumNestedTransactions() > 1 {
```

Comment:
for my understanding: when there are still nested tx here, does it mean some nested tx was not popped bcz of error?

Reply:
in case of processErr == nil, it means that we forgot to call commit somewhere.  in case of processErr != nil, it potentially mean that we trampoline the control flow due to the error

Reply:
added some more comment to the processErr == nil branch